feat: ✨ implemented calculator functionality

de-vasta opened the pull request.

de-vasta committed:
feat: ✨ implemented calculator functionality

DemonWayne commented on src/makeCalculator.js:
```diff
@@ -4,7 +4,39 @@
-  // write code here
+  return {
+    result: 0,
+    add(arg) {
+      this.result += arg;
+
+      return this;
+    },
+    subtract(arg) {
+      this.result -= arg;
+
+      return this;
+    },
+    multiply(arg) {
+      this.result *= arg;
+
+      return this;
+    },
+    divide(arg) {
+      this.result /= arg;
+
+      return this;
+    },
```
Those returns are unnecessary cause we use those functions like callbacks and don't expect them to return anything, so we can remove those returns in my opinion.

de-vasta replied:
You got a point. We do not really need to when we don't call them directly, but within `operate` function

DemonWayne commented on src/makeCalculator.js:
```diff
@@ -4,7 +4,39 @@
+      return this;
+    },
+    divide(arg) {
+      this.result /= arg;
```
I think that raw divide could be harmful and break all calculations if arg is **zero**. We would probably get infinity, and I don't think that's an expected behavior for this calculator. Maybe you can check if arg is **zero** and reset the result?

de-vasta replied:
Ok. In general, as far as I know, calculators tend to give you `NaN`, `Infinity` or some message like "Cannot divide by zero" or so after dividing by zero. You provided quite the interesting approach

DemonWayne reviewed (changes requested):
Looks good, except for the specified problems below 👀

de-vasta committed:
feat: ✨ added division by zero validation and removed method chaining

de-vasta reviewed (commented):
Removed method chaining where it's not used in tests

de-vasta commented on src/makeCalculator.js:
```diff
@@ -8,23 +8,20 @@ function makeCalculator() {
+      if (!arg || arg === 0) {
+        this.reset();
+        throw new Error('Cannot divide by zero');
+      }
```
@DemonWayne, I provided zero handling in division

DemonWayne commented on src/makeCalculator.js:
```diff
@@ -4,7 +4,36 @@
+      this.result *= arg;
+    },
+    divide(arg) {
+      if (!arg || arg === 0) {
```
I think you need either only to check `!arg` or `arg === 0`, like `!arg` by default already checks if the value is `0`

de-vasta replied:
well indeed. `0` is about the same as `false`

DemonWayne reviewed (approved):
Looks good! Specified a minor thing in the if check, but don't think this really matter.

de-vasta committed:
fix: improved calculator error handling and code formatting

- Fixed divide method to check for NaN values instead of just zero
- Removed unused parameter from reset method
- Added consistent spacing between methods for better readability
